src/hotspot/os/posix/os_posix.cpp

```diff
@@ -183,6 +183,17 @@ size_t os::lasterror(char *buf, size_t len) {
   return n;
 }
 
+// Return true if user is running as root.
+bool os::have_special_privileges() {
+  static bool init = false;
+  static bool privileges = false;
+  if (!init) {
+    privileges = (getuid() != geteuid()) || (getgid() != getegid());
+    init = true;
+  }
+  return privileges;
+}
```

Comment:
Could be shortened to:
```
bool os::have_special_privileges() {
  static bool privileges = (getuid() != geteuid()) || (getgid() != getegid());
  return privileges;
}
```

Reply:
Hi Thomas, thanks for the hint - I adjusted the coding.
Best regards, Matthias

Reply:
A static bool which gets evaluated on every call makes no sense!

Reply:
Now I am confused, what class? This is a global function with a function-local static variable.